daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -61,6 +61,17 @@ func New(info logger.Info) (logger.Logger, error) {
 		}
 	}
 
+	var compress = false
+	if maxFiles > 1 {
```

Comment:
Maybe we should always check the compression option and error out if maxFiles is 0 or 1.

Reply:
This was addressing https://github.com/moby/moby/pull/29932#discussion_r149199872.
Log rotation is enabled only if `maxFiles > 1`, do we need to validate this when log rotation is disabled?